src/main/java/io/vertx/ext/mail/impl/MailClientImpl.java

```diff
@@ -138,15 +138,17 @@ private void getConnection(MailMessage message, Handler<AsyncResult<MailResult>>
   }
 
   private void sendMessage(MailMessage email, SMTPConnection conn, Handler<AsyncResult<MailResult>> resultHandler,
-      Context context) {
-    final Handler<AsyncResult<MailResult>> sentResultHandler = result -> {
-      if (result.succeeded()) {
-        conn.returnToPool();
+                           Context context) {
+    Promise<MailResult> sentResultHandler = Promise.promise();
```

Comment:
how about returning a Future that would be created by the Context ?

Reply:
done